### PR TITLE
Add FindUsages external access layer for typescript.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @uniqueiniquity @minestarks  THis is somewhat pressing as it is currently blocking https://github.com/dotnet/roslyn/pull/52854 from being possible.

**Comment by CyrusNajmabadi** [comment]
@sharwell @dotnet/roslyn-ide  please review.

**Comment by CyrusNajmabadi** [comment]
@uniqueiniquity @minestarks the order of operations will be:

1. this PR goes in.  Roslyn defers to your existing IFindUsagesService impl
2. you get our bits.  you then convert your IFindUsagesService impl to instead impl IVSTypeScriptFindUsagesService.  This should be relatively painless.  THe major difference are that the apis pass a cancellation token, and there are wrapper types instead of directly using ours.
3. Remember to *stop* implementing IFindUsagesService.  Our system will pick up your IVSTypeScriptFindUsagesService.
4. Once you get merged, let us know.  We will then remove our bridge interface (called `IFindUsagesServiceRenameOnceTypeScriptMovesToExternalAccess`).

File: src/EditorFeatures/Core/FindUsages/IFindUsagesService.cs [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
-#nullable disable
-
+using System;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.FindUsages;
 using Microsoft.CodeAnalysis.Host;
 
 namespace Microsoft.CodeAnalysis.Editor.FindUsages
 {
+    [Obsolete("Legacy API for TypeScript.  Once TypeScript moves to IVSTypeScriptFindUsagesService", error: false)]
```
Comment (allisonchou) [inline src/EditorFeatures/Core/FindUsages/IFindUsagesService.cs:12]: This comment feels a little incomplete?

File: src/EditorFeatures/Core/FindUsages/IFindUsagesService.cs [hunk 2]
```diff
@@ -24,4 +24,37 @@ internal interface IFindUsagesService : ILanguageService
         /// </summary>
         Task FindImplementationsAsync(Document document, int position, IFindUsagesContext context);
     }
+
+    internal interface IFindUsagesServiceRenameOnceTypeScriptMovesToExternalAccess : ILanguageService
```
Comment (allisonchou) [inline src/EditorFeatures/Core/FindUsages/IFindUsagesService.cs:28]: I feel like a comment here could be useful for clarity (my first instinct was to ask "rename to what?"), or maybe just a link back to this PR/issue?

**Review by allisonchou — APPROVED** [review]

File: src/EditorFeatures/Core/FindUsages/IFindUsagesService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/FindUsages/IFindUsagesService.cs:28]: Gotcha. This is fairly standard with moving something to EA. The point is to make it clear in the name that this is temporary. Once ts moves, we just rename this to the original simple name.

**Pull request merged**